bridges/ethMultiversX/bridgeExecutor.go

```diff
@@ -486,31 +479,42 @@ func (executor *bridgeExecutor) addBatchSCMetadata(ctx context.Context, transfer
 	return transfers, nil
 }
 
-func (executor *bridgeExecutor) addMetadataToTransfer(transfer *bridgeCommon.DepositTransfer, events []*contract.ERC20SafeERC20SCDeposit) *bridgeCommon.DepositTransfer {
+func (executor *bridgeExecutor) addMetadataToTransfer(transfer *bridgeCore.DepositTransfer, events []*contract.ERC20SafeERC20SCDeposit) *bridgeCore.DepositTransfer {
 	for _, event := range events {
 		if event.DepositNonce.Uint64() == transfer.Nonce {
-			transfer.Data = []byte(event.CallData)
-			var err error
-			transfer.DisplayableData, err = ConvertToDisplayableData(transfer.Data)
-			if err != nil {
-				executor.log.Warn("failed to convert call data to displayable data, will alter and call with an empty call data struct",
-					"error", err,
-					"call data", fmt.Sprintf("%+v", emtpyCallData))
-
-				codec := parsers.MultiversxCodec{}
-				transfer.Data = codec.EncodeCallData(emtpyCallData)
-			}
-
+			processData(transfer, []byte(event.CallData))
 			return transfer
 		}
 	}
 
-	transfer.Data = []byte{parsers.MissingDataProtocolMarker} // make([]byte, 0)
+	transfer.Data = []byte{bridgeCore.MissingDataProtocolMarker}
 	transfer.DisplayableData = ""
 
 	return transfer
 }
 
+func processData(transfer *bridgeCore.DepositTransfer, buff []byte) {
+	transfer.Data = buff
+	dataLen := len(transfer.Data)
+	if dataLen == 0 {
+		transfer.Data = []byte{bridgeCore.MissingDataProtocolMarker}
+		transfer.DisplayableData = ""
+		return
+	}
+	if dataLen == 1 && buff[0] == bridgeCore.MissingDataProtocolMarker {
```

Comment:
```suggestion
	// this check is optional, but brings an optimisation to reduce the gas used in case of a bad callData
	if dataLen == 1 && buff[0] == bridgeCore.MissingDataProtocolMarker {
```

Reply:
done